src/org/mockito/internal/configuration/MockAnnotationProcessor.java

```diff
@@ -24,12 +29,50 @@ public Object process(Mock annotation, Field field) {
         } else {
             mockSettings.name(annotation.name());
         }
-        if(annotation.serializable()){
-        	mockSettings.serializable();
+        if (annotation.serializable()) {
+            mockSettings.serializable();
         }
 
-        // see @Mock answer default value
-        mockSettings.defaultAnswer(annotation.answer().get());
+        mockSettings.defaultAnswer(getDefaultAnswer(annotation.customAnswer(), annotation.answer(), field));
         return Mockito.mock(field.getType(), mockSettings);
     }
+
+    private Answer<?> getDefaultAnswer(Class<? extends Answer> customAnswer, Answers defaultAnswer, Field field) {
+
+        if (hasCustomAnswerDefined(customAnswer)) {
+            reportCustomAnswerMisuse(defaultAnswer, field);
+
+            return instantiateCustomAnswer(customAnswer);
+        }
+        return defaultAnswer.get();
+    }
+
+    private Answer<?> instantiateCustomAnswer(Class<? extends Answer> customAnswer) {
+        if(customAnswer.getConstructors().length == 0) {
```

Comment:
Going through your comments and attending to them in my own (planned) Pull Request - not sure what you mean here? Doesn't getConstructor() (with no arguments) check for precisely that public parameter-less constructor? If no such constructor exists, a NoSuchMethodException will be thrown (and then caught by the catch Exception block that you suggested)